### DISCUSSION TITLE
Home for open terraform docs?

### DISCUSSION THREAD

**Discussion by tbugfinder** [opentofu/opentofu#1018] [General]
Hello,

maintenance of terraform-docs slowed down as well as maintainer responsiveness.
I'm wondering whether opentofu could also add a new home for terraform docs?

Thx

**Comment by ghost** [comment]
Hi @tbugfinder maybe the terraform-docs project would be open to more maintainers so the pull requests can get merged again? The project isn't dead, a PR has been merged two days ago, so the maintainers are still around.

**Reply by tbugfinder** [reply]
Well, that was the only activity for quite a while.
There's no reaction a call for adding maintainers, neither new releases...

It's not just about a recent merge...

**Reply by ghost** [reply]
Yes, I can see that they are not very active, but it would be still worth reaching out to them to onboard more maintainers. The project isn't exactly small and forking is a last resort as it splits the existing community.

**Reply by tbugfinder** [reply]
Although a new release was carved out, it still feels very silent with regards to onboarding of additional maintainers.

**Reply by ghost** [reply]
Did someone ask if they would be open to that? Can you link the discussion?

**Reply by rquadling** [reply]
They also have their own [discussion](https://github.com/terraform-docs/terraform-docs/discussions/691) about maintainers.

**Reply by tbugfinder** [reply]
Well, IMHO - that thread is sort of dead therefore I was asking here.

**Reply by rquadling** [reply]
Yep. I included it so it was clear that "work needs to be done".

One of the main issues with terraform-docs is that it relies on a Hashicorp library called [`terraform-config-inspect`](https://github.com/hashicorp/terraform-config-inspect) (see [here](https://github.com/search?q=repo%3Aterraform-docs%2Fterraform-docs%20terraform-config-inspect&type=code)) which is VERY VERY restricted on accepted PRs (the README covers a lot, but also this [discussion](https://discuss.hashicorp.com/t/improving-the-terraform-config-inspect-library/58474/2).

In looking at ALL the OpenToFu code (well, using GitHub's search) for ["terraform-config-inspect"](https://github.com/search?q=org%3Aopentofu+terraform-config-inspect&type=code), there are only 2 mentions of it in code ... as comments ( and a discussion mentions it ... well 2 will by the time I post this comment!)

As terraform-docs is NOT Hashicorps, and the library isn't part of the open source of Terraform, it feels we are in a bit stuck in the middle.

It doesn't look like the library has been forked into OpenToFu's space.

I have made a [fork](https://github.com/rquadling/terraform-docs/tree/master) and added in the functionality I want for terraform-docs (see [PR#675](https://github.com/terraform-docs/terraform-docs/pull/675)).

I'm NOT an expert on any of this sort of thing. Vocal? OK. I'll give you that one!!!

But I would be VERY happy to follow whatever consensus is agreed to.

It may be that the library is really just stop gap that was used by the initial terraform-docs development and that there are better and fuller libraries available that would give the same information. This may mean some refactoring is needed to terraform-docs. With guidance, I could help with that (I'm VERY new to GoLang and so editing existing and working code is different to writing from scratch). But with that, as the tests are already established, the refactoring should be easier if it is required? I hope?!
